animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+          child: new InkWell(
+            onTap: onTap,
+            child: new Image.asset(
+              photo,
+              fit: BoxFit.contain,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* An InkWell wraps the Hero, making it trivial to add a tap
+  gesture to the both the source and destination Heroes.
+* Defining the Material widget with a transparent color
+  enables the image to "pop out" of the background as it
+  flies to its destination.
+* Placing everything in a SizedBox and setting it's width
+  to the passed-in value, determines the size
+  of the image at the end of the animation.
+* Setting the Image's `fit` property to `BoxFit.contain`,
+  ensures that the image is as large as possible during the
+  transition while fitting into the target box.
+
+### SharedElementTransition class
+
+The SharedElementTransition class creates the source and destination
+PhotoHeroes, and sets up the transition.
+
+Here's the code:
+
+<!-- skip -->
+{% prettify dart %}
+class SharedElementTransition extends StatelessWidget {
+  Widget build(BuildContext context) {
+    [[highlight]]timeDilation = 5.0; // 1.0 means normal animation speed.[[/highlight]]
+
+    return new Scaffold(
+      appBar: new AppBar(
+        title: const Text('Shared Element Transition'),
+      ),
+      body: new Center(
+        [[highlight]]child: new PhotoHero([[/highlight]]
+          photo: 'images/flippers-alpha.png',
+          width: 300.0,
+          [[highlight]]onTap: ()[[/highlight]] {
+            [[highlight]]Navigator.of(context).push(new MaterialPageRoute<Null>([[/highlight]]
+              [[highlight]]builder: (BuildContext context)[[/highlight]] {
+                return new Scaffold(
+                  appBar: new AppBar(
+                    title: const Text('Flippers Page'),
+                  ),
+                  body: new Container(
+                    // The blue background emphasizes that it's a new route.
+                    color: Colors.lightBlueAccent,
+                    padding: const EdgeInsets.all(16.0),
+                    alignment: Alignment.topLeft,
+                    [[highlight]]child: new PhotoHero([[/highlight]]
+                      photo: 'images/flippers-alpha.png',
+                      width: 100.0,
+                      [[highlight]]onTap: ()[[/highlight]] {
+                        [[highlight]]Navigator.of(context).pop();[[/highlight]]
+                      },
+                    ),
+                  ),
+                );
+              }
+            ));
+          },
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* When the user taps the InkWell containing the source Hero,
+  the code creates a new MaterialPageRoute (in `onTap()`) and pushes
+  it to the Navigator’s stack, triggering the animation.
+* The anonymous builder for the new route creates a Container with the
```

Comment:
Referring to this as anonymous isn't wrong but it doesn't convey very much. You're referring to the destination MaterialRoute's page `builder` function.